File: Projetos/Alura/Kotlin/Introducao/bytebank/src/main/kotlin/Conta.kt
class Conta(titular: String, numero: Int) {
    private var titular: String? = null
    private var numero: Int? = null

    private var saldo: Double = 0.0
        set(valor){
            if(valor > 0){field = valor}
        }

    fun saca(valor: Double) {
        when {
            saldo >= valor -> saldo -= valor
        }
    }

    fun deposita(valor: Double) {
        when {
            valor > 0 -> saldo += valor
        }
    }

    fun transfere(valor: Double, contaDestino: Conta): Boolean {
        when {
            saldo >= valor -> {
                saldo -= valor
                contaDestino.deposita(valor)
                return true
            }
            else -> return false
        }
    }

    fun verExtrato() {
        println("Saldo ${titular}: ${saldo}")
    }

}

File: Projetos/Alura/Curso1/conta.py
class Conta:

    def __init__(self, numero, titular, saldo, limite):
        print("Construindo objeto... {}".format(self))
        self.__numero = numero
        self.__titular = titular
        self.__saldo = saldo
        self.__limite = limite

    def extrato(self):
        print("Saldo: R${} do titular {}".format(self.__saldo, self.__titular))

    def deposita(self, valor):
        self.__saldo += valor

    def saca(self, valor):
        self.__saldo -= valor

    def transfere(self, valor, destino):
        if(self.__saldo > valor):
            self.saca(valor)
            destino.deposita(valor)
            print("Valor de R${} para {} feito com sucesso!".format(valor, destino.__titular))
        else:
            print("Você não tem saldo o suficiente.")
